src/OpenTelemetry.Instrumentation.Wcf/Implementation/InstrumentedCommunicationObject.cs

```diff
@@ -41,7 +42,7 @@ CommunicationState ICommunicationObject.State
         get { return this.Inner.State; }
     }
 
-    protected ICommunicationObject Inner { get; }
+    protected T Inner { get; }
```

Comment:
@repl-chris I just pushed a little bit of refactoring so that we can expose a strongly-typed "Inner" on the base class. That gets rid of the need for `[private|protected] new [ChannelType] Inner { get => [ChannelType]base.Inner; }` we had been using in all the derived classes.

Reply:
Nice....looks good 👍